python/dask_cudf/dask_cudf/io/tests/test_s3.py

```diff
@@ -59,11 +63,11 @@ def s3_base(endpoint_port):
 
         # Launching moto in server mode, i.e., as a separate process
         # with an S3 endpoint on localhost
-        endpoint_uri = f"http://127.0.0.1:{endpoint_port}/"
 
-        proc = subprocess.Popen(
-            shlex.split(f"moto_server s3 -p {endpoint_port}"),
-        )
+        endpoint_uri = f"http://{endpoint_ip}:{endpoint_port}/"
+
+        server = ThreadedMotoServer(ip_address=endpoint_ip, port=endpoint_port)
+        server.start()
```

Comment:
@galipremsagar One more note: this `server.start()` call appears to block until the server is ready. From local testing, it appears that the "timeout" code below may not be necessary.

```python
>>> server.start(); print(requests.get(f"http://{endpoint_ip}:{endpoint_port}/").ok); server.stop()
Starting a new Thread with MotoServer running on 0.0.0.0:5000...
127.0.0.1 - - [10/May/2022 14:35:25] "GET / HTTP/1.1" 200 -
True
```

In that case, we can replace this with:
```python
server.start()
yield endpoint_uri
server.stop()
```

Reply:
Cool, that makes sense. I was skeptical about removing it too, but this confirms.

Reply:
Removed the code.

Reply:
This also removes the dependency on `requests`! 💡

Reply:
> This also removes the dependency on `requests`! 💡

Nice catch! Done